### PR TITLE
DAOS-7595 test: Automation - Verify automatic object class selection

### PR DESCRIPTION
**Pull request by shimizukko**
Create a container with different RF and OC combinations. We expect some
combinations to fail due to the restriction set by DAOS.

Skip-unit-tests: true
Test-tag: oc_restrictions
Signed-off-by: Makito Kano <makito.kano@intel.com>

### CODE REVIEW

File: src/tests/ftest/container/auto_oc_selection.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            ("rf:3", "RP_3G1", True),
+            ("rf:3", "RP_4G1", False),
+            ("rf:3", "RP_5G1", False),
+            ("rf:4", "S1", True),
+            ("rf:4", "RP_2G1", True),
+            ("rf:4", "RP_3G1", True),
+            ("rf:4", "RP_4G1", True),
+            ("rf:4", "RP_5G1", False)
+        ]
+        self.container = []
```
Comment (daltonbohning) [inline src/tests/ftest/container/auto_oc_selection.py:67]: Question - I don't think `self.container` is actually being used. I think we should either remove this or append the created containers below.

File: src/tests/ftest/container/auto_oc_selection.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                pool=self.pool, daos_command=self.get_daos_command())
+            cont.get_params(self)
+            cont.properties.update(properties)
+            cont.oclass.update(oclass)
+
+            try:
+                self.log.debug(
+                    "Create container with RF = %s, OC = %s", properties,
+                    oclass)
+                cont.create()
```
Comment (daltonbohning) [inline src/tests/ftest/container/auto_oc_selection.py:86]: Suggestion/Optional - Could we use `TestWithServers.get_container()`, but with added params for `properties` and `oclass` so other tests can benefit?

File: src/tests/ftest/container/auto_oc_selection.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    """Automatic object class selection test.
+
+    Test Class Description:
+    Create a container with different RF and OC combinations. We expect some
+    combinations to fail due to the restriction set by DAOS. (This is more like
+    auto OC "restriction" rather than auto OC "selection".)
+
+    :avocado: recursive
+    """
+    def test_oc_restrictions(self):
```
Comment (daltonbohning) [inline src/tests/ftest/container/auto_oc_selection.py:22]: Question - Is there a reason for mixing "restriction" and "selection"? Could we simplify and use just one term?
- Filename `auto_oc_selection.py`
- Class name `AutoOCSelectionTest`
- Test method name `test_oc_restrictions`
- Test tag `oc_restrictions`

File: src/tests/ftest/container/auto_oc_selection.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        Create a container with RF 0 to 4 - OC S1, RP_2G1, RP_3G1, RP_4G1, and
+        RP_5G1.
+
+        Use Cases:
+        Create a pool. Then create each container and verify against its
+        failure expected state.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=vm
+        :avocado: tags=container,oc_restrictions
```
Comment (daltonbohning) [inline src/tests/ftest/container/auto_oc_selection.py:35]: Suggestion/Minor/Optional - the "features" tag is usually on a separate line
```suggestion
        :avocado: tags=all,full_regression
        :avocado: tags=vm
        :avocado: tags=container
        :avocado: tags=oc_restrictions
```

**Review by dinghwah — APPROVED** [review]
LGTM

**Review by daltonbohning — APPROVED** [review]
LGTM. Thanks for the updates!

File: src/tests/ftest/container/auto_oc_selection.py (hunk 1 above)
Comment (shimizukko) [reply src/tests/ftest/container/auto_oc_selection.py:68]: Done. Thanks for pointing it out.

File: src/tests/ftest/container/auto_oc_selection.py (hunk 2 above)
Comment (shimizukko) [reply src/tests/ftest/container/auto_oc_selection.py]: Done

File: src/tests/ftest/container/auto_oc_selection.py (hunk 3 above)
Comment (shimizukko) [reply src/tests/ftest/container/auto_oc_selection.py]: Done

File: src/tests/ftest/container/auto_oc_selection.py (hunk 4 above)
Comment (shimizukko) [reply src/tests/ftest/container/auto_oc_selection.py]: Done

**Pull request merged**